andela/zinnia-ah-backend
# #164797166 Sets up roles and access control

#### What does this PR do?
sets up roles and access control

#### Description of Task to be completed?
- [x] add role column to users table
- [x] setup middleware to lock routes to admin only
- [x] setup endpoints to grant and revoke admin access

#### How should this be manually tested?
- Clone the repo,
- Run ```npm install``` to install all dependencies,
- Create .env file using the .env.example as a guide and enter you environment variables
- Run `sequelize db:create`  
- Run `sequelize db:migrate` 
- Run `sequelize db:seed:all`   
- Run npm run start:dev,
- send a POST request to `localhost:3000/api/v1/auth/login` using `igbominadeveloper@ah.com` and password `password1`
- On Postman, send a GET request to localhost:3000/api/v1/roles
- You should get the screenshot below

#### Any background context you want to provide?
Some routes should be accessible to an admin only

#### What are the relevant pivotal tracker stories? (if applicable)
[#164797166](https://www.pivotaltracker.com/story/show/164797166)

#### Screenshots
<img width="1440" alt="Screenshot 2019-04-14 at 6 02 20 PM" src="https://user-images.githubusercontent.com/12516195/56115140-a2879280-5f5a-11e9-834a-3d3efdf2cdeb.png">

![Roles and access control](https://user-images.githubusercontent.com/12516195/56117257-9ce07b80-5f5f-11e9-9a14-eeea501e8e65.png)
